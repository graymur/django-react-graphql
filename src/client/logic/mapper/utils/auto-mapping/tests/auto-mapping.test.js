const autoMapping = require('../index').default
const fromWithConfig = require('./data/from-fields-with-auto-mapping-config')
const toWithConfig = require('./data/to-fields-with-auto-mapping-config')
const fromWithoutConfig = require('./data/from-fields-without-auto-mapping-config')
const toWithoutConfig = require('./data/to-fields-without-auto-mapping-config')

test('maps based on config', () => {
	const result = autoMapping(fromWithConfig, toWithConfig)
	expect(result).toMatchSnapshot()
})

test('maps based on fields names', () => {
	const result = autoMapping(fromWithoutConfig, toWithoutConfig, {
		matchByFieldsNames: true,
	})
	expect(result).toMatchSnapshot()
})

test('maps both', () => {
	const result = autoMapping(
		[...fromWithConfig, ...fromWithoutConfig],
		[...toWithConfig, ...toWithoutConfig],
		{
			matchByFieldsNames: true,
		},
	)
	expect(result).toMatchSnapshot()
})
