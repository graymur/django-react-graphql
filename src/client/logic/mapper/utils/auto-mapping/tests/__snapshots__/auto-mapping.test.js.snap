// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps based on config 1`] = `
Array [
  Object {
    "constraintsMessages": Array [],
    "from": "from_firstName",
    "to": "to_firstName",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_lastName",
    "to": "to_lastName",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_email",
    "to": "to_email",
  },
]
`;

exports[`maps based on fields names 1`] = `
Array [
  Object {
    "constraintsMessages": Array [],
    "from": "from_first_space_Name",
    "to": "to_first_Name",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_last_space_Name",
    "to": "to_last-Name",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "fromemail",
    "to": "to_e.mail",
  },
]
`;

exports[`maps both 1`] = `
Array [
  Object {
    "constraintsMessages": Array [],
    "from": "from_firstName",
    "to": "to_firstName",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_lastName",
    "to": "to_lastName",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_email",
    "to": "to_email",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_first_space_Name",
    "to": "to_first_Name",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "from_last_space_Name",
    "to": "to_last-Name",
  },
  Object {
    "constraintsMessages": Array [],
    "from": "fromemail",
    "to": "to_e.mail",
  },
]
`;
